src/containers/collaborations/components/AddOrganizationsToCollaborationModal.js

```diff
@@ -0,0 +1,127 @@
+/*
+ * @flow
+ */
+
+import React, { useMemo, useState } from 'react';
+
+import { List, Map } from 'immutable';
+import { CollaborationsApiActions } from 'lattice-sagas';
+import { ActionModal, Label, Select } from 'lattice-ui-kit';
+import { useRequestState } from 'lattice-utils';
+import { useDispatch, useSelector } from 'react-redux';
+import { RequestStates } from 'redux-reqseq';
+import type { Organization, UUID } from 'lattice';
+import type { RequestState } from 'redux-reqseq';
+
+import { ModalBody, StackGrid } from '../../../components';
+import { resetRequestStates } from '../../../core/redux/actions';
+import { COLLABORATIONS } from '../../../core/redux/constants';
+import { selectOrganizations } from '../../../core/redux/selectors';
+import type { ReactSelectOption } from '../../../types';
+
+const { ADD_ORGANIZATIONS_TO_COLLABORATION, addOrganizationsToCollaboration } = CollaborationsApiActions;
+
+const AddOrganizationsToCollaborationModal = ({
+  collaborationId,
+  isVisible,
+  onClose,
+  participatingOrganizations
+} :{
+  collaborationId :UUID;
+  isVisible :boolean;
+  onClose :() => void;
+  participatingOrganizations :List;
+}) => {
+
+  const dispatch = useDispatch();
+
+  const [collaborationOrganizations, setCollaborationOrganizations] = useState([]);
+
+  const organizations :Map<UUID, Organization> = useSelector(selectOrganizations());
+  const addOrgsToCollaborationRS :?RequestState = useRequestState([COLLABORATIONS, ADD_ORGANIZATIONS_TO_COLLABORATION]);
+
+  const options = useMemo(() => {
+    const selectOptions = [];
+    organizations.forEach((org) => {
+      if (!participatingOrganizations.includes(org.id)) {
+        selectOptions.push({
+          key: org.id,
+          label: org.title,
+          value: org.id,
+        });
+      }
+    });
+
+    return selectOptions;
+  }, [organizations, participatingOrganizations]);
+
+  const handleOnChange = (orgOptions :ReactSelectOption<Organization>[]) => {
+    const orgIds = orgOptions.map((org) => org.value);
+    setCollaborationOrganizations(orgIds);
+  };
+
+  const handleOnClickPrimary = () => {
+    dispatch(
+      addOrganizationsToCollaboration({
+        collaborationId,
+        organizationIds: collaborationOrganizations
+      })
+    );
+  };
+
+  const handleOnClose = () => {
+    onClose();
+    setTimeout(() => {
+      dispatch(resetRequestStates([ADD_ORGANIZATIONS_TO_COLLABORATION]));
+    }, 1000);
```

Comment:
Instead of using a timeout to avoid ui flicker when closing the modal, try using the `ResetOnUnmount` wrapper component. Also looking back on the usage of ActionModals and similar components, I'm not sure why the Pending state is always omitted...